File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnSurfaceVariantLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnSurfaceVariantLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl6
            AppTheme.STANDARD -> R.color.cstl6
            AppTheme.BLUE -> R.color.cbl6
            AppTheme.GREEN -> R.color.cgl6
            AppTheme.RED -> R.color.crl6
            AppTheme.NOTHING -> R.color.cnol6
        }
    }
}

File: lib/layer2_core/light/e_on_surface_variant_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnSurfaceVariantLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl6;
      case 'standard': return cstl6;
      case 'blue': return cbl6;
      case 'green': return cgl6;
      case 'red': return crl6;
      case 'nothing': return cnol6;
      default: return cstl6;
    }
  }
}
